src/java/org/apache/cassandra/tools/nodetool/stats/DataPathsPrinter.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.tools.nodetool.stats;
+
+import java.io.PrintStream;
+import java.util.List;
+import java.util.Map;
+
+public class DataPathsPrinter<T extends StatsHolder>
+{
+    public static <T extends StatsHolder> StatsPrinter<T> from (String format)
```

Comment:
@rtib by the way this whole method might be changed to this:

    public static StatsPrinter<DataPathsHolder> from (String format)
    {
        if (format.equals("json"))
            return new StatsPrinter.JsonPrinter<>();
        if (format.equals("yaml"))
            return new StatsPrinter.YamlPrinter<>();

        return new DefaultPrinter();
    }

You do not need to do it with generics.

Then in nodetool command, you need to change it like this:

        DataPathsHolder holder = new DataPathsHolder(probe, tableNames);
        StatsPrinter<DataPathsHolder> printer = DataPathsPrinter.from(outputFormat);
        printer.print(holder, probe.output().out);

It will be nicely done without any casts or whatever.